### PR TITLE
Extend password validation to include more special characters

### PR DESCRIPTION
**Pull request by ogeeDeveloper**
This update modifies the password validation regex to allow additional special characters, including the period (.). Previously, the validation logic restricted passwords to a specific set of special characters, inadvertently excluding valid and commonly used characters like the period. This change ensures that passwords with a broader range of special characters, including periods, pass validation. It maintains the requirement for passwords to be at least 8 characters long and to include at least one uppercase letter, one lowercase letter, one number, and one special character.

### CODE REVIEW

**Comment by coltonehrman** [comment]
@ogeeDeveloper it looks like some formatting happened and hid your changes. What was changed in this PR?

**Comment by coltonehrman** [comment]
@ogeeDeveloper Also, is it possible to extract the RegEx being used and write a very simple test case with it to validate.

**Comment by ogeeDeveloper** [comment]
> @ogeeDeveloper it looks like some formatting happened and hid your changes. What was changed in this PR?

This was changed:
password: z.string()
  .min(8, { message: "Password must be at least 8 characters long" })
  .regex(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&.])[A-Za-z\d@$!%*?&.]{8,}$/, { message: "Password must include uppercase, lowercase, number, and special character" }),
The regex was modified. Also it should be possible to extract the regex and write a test for it

**Comment by coltonehrman** [comment]
> > @ogeeDeveloper it looks like some formatting happened and hid your changes. What was changed in this PR?
> 
> This was changed: password: z.string() .min(8, { message: "Password must be at least 8 characters long" }) .regex(/^(?=._[a-z])(?=._[A-Z])(?=._\d)(?=._[@$!%_?&.])[A-Za-z\d@$!%_?&.]{8,}$/, { message: "Password must include uppercase, lowercase, number, and special character" }), The regex was modified. Also it should be possible to extract the regex and write a test for it

Can we include some basic tests in this PR?

**Comment by ogeeDeveloper** [comment]
Yes, I'm working on those tests now

**Comment by coltonehrman** [comment]
Closes #21

**Comment by coltonehrman** [comment]
Closes #36

**Comment by coltonehrman** [comment]
Closes #33

**Comment by coltonehrman** [comment]
@ogeeDeveloper I went ahead and formatted `main`, if you can sync your branch, the diff should be a lot cleaner

File: tests/passwordValidation.test.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+const test = require("node:test");
+const assert = require("node:assert");
+
+const passwordRegex =
+  /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&.])[A-Za-z\d@$!%*?&.]{8,}$/;
+
+test("password validation", async (t) => {
+  await t.test("should accept a valid password", async () => {
```
Comment (coltonehrman) [inline tests/passwordValidation.test.js:8]: Do we need `async/await` at all? These should all be sync

**Pull request merged**